tests/bluetooth/ctrl_isoal/src/main.c

```diff
@@ -20,1517 +20,7747 @@
+static isoal_status_t sink_sdu_emit_test(const struct isoal_sink *sink_ctx,
+					 const struct isoal_sdu_produced *valid_sdu)
 {
-	/* Assume SDU buffer len 5 (sdu_buf_len=5) */
-
-	clear_sdu_buf();
-	construct_pdu_unframed(PDU_BIS_LLID_START_CONTINUE, &testdata[0], 10,
-			       payload_number++, 10000, 0);
-	/* Test recombine */
-	pdu_ref = &testdata[0];
-	err = isoal_rx_pdu_recombine(sink_hdl, &pdu_meta);
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-	/* should only see 5 written as two SDUs 5 each has been generated */
-	zassert_equal(sink->sdu_production.sdu_written, 5, "written=%u",
-		      sink->sdu_production.sdu_written);
-	zassert_equal(sink->sdu_production.sdu_status, ISOAL_SDU_STATUS_VALID, "sdu_status=0x%x",
-		      sink->sdu_production.sdu_status);
+	debug_trace_func_call(__func__, "IN");
+
+	debug_print_rx_sdu(sink_ctx, ((struct rx_sdu_frag_buffer *)valid_sdu->contents.dbuf)->sdu);
+
+	ztest_check_expected_value(sink_ctx);
+	ztest_check_expected_value(sink_ctx->sdu_production.sdu_state);
+	ztest_check_expected_value(sink_ctx->sdu_production.sdu_written);
+	ztest_check_expected_value(valid_sdu->status);
+	ztest_check_expected_value(valid_sdu->timestamp);
+	ztest_check_expected_value(valid_sdu->seqn);
+	ztest_check_expected_value(valid_sdu->contents.dbuf);
+	ztest_check_expected_value(valid_sdu->contents.size);
+
+	return ztest_get_return_value();
 }
 
 /**
- * @brief   Test create and destroy sink
+ * Callback test fixture to be provided for RX sink creation. Writes provided
+ * data into target SDU buffer.
+ * @param  dbuf        SDU buffer (Includes current write location field)
+ * @param  pdu_payload Current PDU being reassembled by ISO-AL
+ * @param  consume_len Length of data to transfer
+ * @return             Status of the operation
  */
-void test_sink_create_destroy(void)
+static isoal_status_t sink_sdu_write_test(void *dbuf,
+					  const uint8_t *pdu_payload,
+					  const size_t consume_len)
 {
-	isoal_sink_handle_t hdl[CONFIG_BT_CTLR_ISOAL_SINKS];
-	struct isoal_sink_config *config_ptr;
-
-	err = isoal_init();
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
-
-	err = isoal_reset();
-	zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
+	debug_trace_func_call(__func__, "IN");
 
-	uint8_t dummy_role = BT_CONN_ROLE_CENTRAL;
+#if defined(DEBUG_TEST)
+	zassert_not_null(dbuf, "");
+	zassert_not_null(pdu_payload, "");
 
-	for (int i = 0; i < CONFIG_BT_CTLR_ISOAL_SINKS; i++) {
-		/* Create a sink based on global parameters */
-		err = isoal_sink_create(handle, dummy_role, framed,
-					burst_number, flush_timeout,
-					sdu_interval, iso_interval,
-					stream_sync_delay, group_sync_delay,
-					sink_sdu_alloc_test, sink_sdu_emit_test,
-					sink_sdu_write_test, &hdl[i]);
-		zassert_equal(err, ISOAL_STATUS_OK, "err=0x%02x", err);
+	struct rx_sdu_frag_buffer *rx_sdu_frag_buf;
 
-		isoal_sink_enable(hdl[i]);
+	rx_sdu_frag_buf = (struct rx_sdu_frag_buffer *)dbuf;
```

Comment:
Why isn't the function taking a `struct rx_sdu_frag_buffer *` instead of a `void *` buffer?